airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -731,24 +733,42 @@ def __init__(self, replication_key: str, stream_slice_step: str = "P30D", **kwar
         super().__init__(**kwargs)
         self.replication_key = replication_key
         self._stream_slice_step = stream_slice_step
+        self._cursor = None
+
+    @property
+    def cursor(self) -> Optional[Cursor]:
+        if not self._cursor:
+            raise ValueError("Cursor should be set at this point")
+        return self._cursor
+
+    @cursor.setter
+    def cursor(self, cursor: Cursor):
+        self._cursor = cursor
 
     def stream_slices(
         self, *, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
     ) -> Iterable[Optional[Mapping[str, Any]]]:
-        now = pendulum.now(tz="UTC")
-        assert LOOKBACK_SECONDS is not None and LOOKBACK_SECONDS >= 0
-
-        initial_date = self.get_start_date_from_state(stream_state) - pendulum.Duration(seconds=LOOKBACK_SECONDS)
-        slice_start = initial_date
-        while slice_start < now:
-            slice_end = slice_start + self.stream_slice_step
-            self._slice = {
-                "start_date": slice_start.isoformat(timespec="milliseconds"),
-                "end_date": min(slice_end, now).isoformat(timespec="milliseconds"),
-            }
-            yield self._slice
-
-            slice_start += self.stream_slice_step
+        if self._cursor:
+            for slice_start, slice_end in self._cursor.generate_slices():
+                yield {
+                    "start_date": slice_start.isoformat(timespec="milliseconds"),
+                    "end_date": slice_end.isoformat(timespec="milliseconds"),
+                }
+        else:
+            now = pendulum.now(tz="UTC")
+            assert LOOKBACK_SECONDS is not None and LOOKBACK_SECONDS >= 0
```

Comment:
why?

Reply:
I don't know. It indeed does not make sense as we are fully owner of `LOOKBACK_SECONDS`. This is code that was there before.

Reply:
Adding to that, some tests were patching this constant with invalid values. I just removed them as it is not a real case scenario